packages/backend-api/src/Controller/V1/Product/ProductApiDataValidator.php

```diff
@@ -0,0 +1,168 @@
+                        'allowExtraFields' => false,
+                    ])
+                ),
+                'hidden' => new Constraints\Optional(
+                    new Constraints\Type([
+                        'type' => 'bool',
+                        'message' => 'The value {{ value }} is not a valid {{ type }}.',
+                    ])
+                ),
+                'sellingDenied' => new Constraints\Optional(
+                    new Constraints\Type([
+                        'type' => 'bool',
+                        'message' => 'The value {{ value }} is not a valid {{ type }}.',
+                    ])
+                ),
+                'sellingFrom' => new Constraints\Optional(
+                    new Constraints\DateTime([
+                        'format' => DateTime::ATOM,
+                        'message' => 'The value {{ value }} is not a valid DateTime::ATOM format.',
+                    ])
+                ),
+                'sellingTo' => new Constraints\Optional(
+                    new Constraints\DateTime([
+                        'format' => DateTime::ATOM,
+                        'message' => 'The value {{ value }} is not a valid DateTime::ATOM format.',
+                    ])
+                ),
+                'catnum' => new Constraints\Optional(
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+                    ])
+                ),
+                'ean' => new Constraints\Optional(
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+                    ])
+                ),
+                'partno' => new Constraints\Optional(
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'The value {{ value }} cannot be longer then {{ limit }} characters',
+                    ])
+                ),
+                'shortDescription' => new Constraints\Optional(
+                    new Constraints\Collection([
+                        'fields' => $descriptionFields,
+                        'allowMissingFields' => true,
+                        'allowExtraFields' => false,
+                    ])
+                ),
+                'longDescription' => new Constraints\Optional(
+                    new Constraints\Collection([
+                        'fields' => $descriptionFields,
+                        'allowMissingFields' => true,
+                        'allowExtraFields' => false,
+                    ])
+                ),
+            ],
+            'allowExtraFields' => false,
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    protected function createNameConstrains(): array
```

Comment:
typo: (https://www.quora.com/What-is-the-difference-between-constraint-and-constrains)
```diff
-    protected function createNameConstrains(): array
+    protected function createNameConstraints(): array
```

Also, method is named createConstraints, but internally working with fields and return value is passed to fields too. Maybe rename it completely could be nice, but I'm not forcing it.

Reply:
renamed to
* `createNameConstraintDefinition`
* `createDescriptionConstraintDefinition`